src/kernel.ts

```diff
@@ -625,13 +610,14 @@ class Kernel {
         // if that fails, then assume the kernel is dead,
         // otherwise just follow the typical websocket closed
         // protocol.
-        this.getInfo().then(function() {
-          this._ws_closed(ws_host_url, false);
-        }, function() {
-          this._kernel_dead();
+        this.getInfo().then(() => {
+          this._wsClosed(ws_host_url, false);
+        }, () => {
+          this._kernelDead();
```

Comment:
This would have been a runtime failure because `this._ws_closed` and `this._kernel_dead` do not exist. The compiler didn't catch it because an anonymous `function` was used instead of an arrow function. This is a double-oops, because even if those functions did exists, the `this` binding would be incorrect and it would still fail.